src/unix/fs.c

```diff
@@ -698,10 +701,90 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_utime(uv_fs_t* req) {
+#if defined(__linux__)
+  /* utimesat() has nanosecond resolution but we stick to microseconds
+   * for the sake of consistency with other platforms.
+   */
+  static int no_utimesat;
+  struct timespec ts[2];
+  struct timeval tv[2];
+  int r;
+
+  if (no_utimesat)
```

Comment:
I don't think we need a fallback anymore as `utimensat` was added in kernel 2.6.22 and currently we support kernels starting with `2.6.32`, so I think you could merge linux with sun and aix71. Also `Freebsd` support `utimensat` starting with version `10.3` but we still support older versions.

Reply:
> Also Freebsd support utimensat starting with version 10.3 but we still support older versions.

I'm not really sure how to detect that... Are the platform define symbols string versions or something?